fuel/src/test/kotlin/com/github/kittinunf/fuel/RoutingTest.kt

```diff
@@ -163,4 +178,37 @@ class RoutingTest: BaseTestCase() {
 
         assertThat(string, containsString("42"))
     }
+
+    @Test
+    fun httpRouterPostBinaryBody() {
+        var request: Request? = null
+        var response: Response? = null
+        var data: Any? = null
+        var error: FuelError? = null
+
+        val paramValue = "42"
+
+        manager.request(TestApi.PostBinaryBodyTest(paramValue)).responseString { req, res, result ->
+            request = req
+            response = res
+
+            val (d, err) = result
+            data = d
+            error = err
+        }
+
+        val string = data as String
+
+        assertThat(request, notNullValue())
+        assertThat(response, notNullValue())
+        assertThat(error, nullValue())
+        assertThat(data, notNullValue())
+
+        val statusCode = HttpURLConnection.HTTP_OK
+        assertThat(response?.statusCode, isEqualTo(statusCode))
+
+        val json = JSONObject(string)
+        val bytes = Base64.decode(json.optString("data"), Base64.DEFAULT)
+        assertThat(String(bytes), isEqualTo(paramValue))
```

Comment:
Lastly, could you please test this against the response body coming from the server not only just the paramValue. I think httpbin return data back to us as well.